src/app/themes.tsx

```diff
@@ -10,53 +10,121 @@ import {
     createTheme as createMaterialTheme,
 } from "@mui/material";
 
-type CustomTheme = {
-    foregroundColor: string;
-    backgroundColor: string;
-    primaryForegroundColor: string;
-    primaryBackgroundColor: string;
-    secondaryForegroundColor: string;
-    secondaryBackgroundColor: string;
-    accentBackgroundColor: string;
-    accentForegroundColor: string;
-    surfaceForegroundColor: string;
-    surfaceBackgroundColor: string;
-    errorColor: string;
+export type standardPalette = {
+    background: string[];
+    foreground: string[];
+    largeForeground: string[];
 };
 
-const lightTheme: CustomTheme = {
-    foregroundColor: "#000000",
-    backgroundColor: "#eeeeee",
-    primaryForegroundColor: "#000000",
-    primaryBackgroundColor: "#63a036",
-    secondaryForegroundColor: "#ffffff",
-    secondaryBackgroundColor: "#066940",
-    accentBackgroundColor: "#609fd3",
-    accentForegroundColor: "#000000",
-    surfaceForegroundColor: "#000000",
-    surfaceBackgroundColor: "#ffffff",
-    errorColor: "#ff624e",
+export type rainbowPalette = {
+    red: standardPalette;
+    orange: standardPalette;
+    yellow: standardPalette;
+    lightGreen: standardPalette;
+    darkGreen: standardPalette;
+    blue: standardPalette;
+    purple: standardPalette;
+    grey: standardPalette;
+    brown: standardPalette;
 };
 
-const darkTheme: CustomTheme = {
-    foregroundColor: "#ffffff",
-    backgroundColor: "#1a1a1a",
-    primaryForegroundColor: "#ffffff",
-    primaryBackgroundColor: "#066940",
-    secondaryForegroundColor: "#000000",
-    secondaryBackgroundColor: "#63a036",
-    accentBackgroundColor: "#1b629c",
-    accentForegroundColor: "#ffffff",
-    surfaceForegroundColor: "#ffffff",
-    surfaceBackgroundColor: "#161414",
-    errorColor: "#ff624e",
+export type customTheme = {
+    light: boolean;
+    main: standardPalette & { border: string };
+    primary: standardPalette;
+    error: string;
+    shadow: string;
+    rainbow: rainbowPalette;
+};
+
+export const lightTheme: customTheme = {
+    light: true,
+    main: {
+        background: ["#fdfdfd", "#eeeeee", "#eeeeee", "#eeeeee"],
+        foreground: ["#000000", "#2d2d2d", "#383838", "#4d4d4d"],
+        largeForeground: ["#000000", "#2d2d2d", "#383838", "#4d4d4d"],
```

Comment:
Also don't see why we need 4 colours for everything? Like especially for a background colour we maybe need one or two, but the more colours the less consistency will come when people use whatever shade they fancy I would imagine. Obviously if there is a reason fair enough.

Reply:
You are right -  I reduced it to three instead of four
I still keep it as three but not two because I need one for grey text (can refer to login panel)